pyslackersweb/cli/migrate.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+import pathlib
+import glob
+
+from typing import List
+from dataclasses import dataclass, field
+
+import asyncpg
+
+from pyslackersweb.util.log import ContextAwareLoggerAdapter
+
+
+logger = ContextAwareLoggerAdapter(logging.getLogger(__name__))
+
+
+@dataclass(frozen=True, order=True)
+class Migration:
+    version: int
+    path: pathlib.Path = field(compare=False)
+
+    async def execute(self, db):
+        with open(self.path, mode="r") as f:
+            async with db.acquire() as connection:
+                await connection.execute(f.read())
+                await connection.execute("UPDATE metadata SET version = $1", self.version)
```

Comment:
I think we need this system to be slightly more advanced. Two things come to mind that would be useful:

1. Using a version table that has each version as an entry with the applied time and item 2
2. Add a hash column which can ensure the hashes of migrations/versions haven’t changed

Reply:
Maybe on a future PR :)

Reply:
Did it now actually, I just don't store a hash but the executed sql.